TRModelTransporter/Packing/AbstractTexturePacker.cs

```diff
@@ -174,13 +174,9 @@ protected IEnumerable<int> GetMeshTextureIndices(List<TRMesh> meshes)
         ISet<int> textureIndices = new SortedSet<int>();
         foreach (TRMesh mesh in meshes)
         {
-            foreach (TRFace4 rect in mesh.TexturedRectangles)
+            foreach (TRMeshFace face in mesh.TexturedFaces)
             {
-                textureIndices.Add(rect.Texture);
-            }
-            foreach (TRFace3 tri in mesh.TexturedTriangles)
-            {
-                textureIndices.Add(tri.Texture);
+                textureIndices.Add(face.Texture);
             }
         }
         return textureIndices;
```

Comment:
This function could be
```
protected static IEnumerable<int> GetMeshTextureIndices(List<TRMesh> meshes) 
    => meshes.SelectMany(m => m.TexturedFaces.Select(f => (int)f.Texture)).ToImmutableSortedSet();
```
Might want to test performance with a big level though, not sure if it will make a big temporary list.

Reply:
Thanks, performance seems good - if anything, better. Set it to `ISet` and it only enumerates once. Just put it straight into the caller as it's only used there. More deleted lines is nice.

This class is another candidate for rewrite to be honest; the whole packing/transport library will get some TLC soon.